tests/modules/systemd/user-config.nix

```diff
@@ -0,0 +1,26 @@
+{ pkgs, ... }:
+
+{
+  systemd.user.settings.Manager = {
+    LogLevel = "debug";
+    DefaultCPUAccounting = true;
+    DefaultEnvironment = {
+      TEST = "abc";
+      PATH = "/bin:/sbin:/some where";
+    };
+  };
+
+  nmt.script = ''
+    userConf=home-files/.config/systemd/user.conf
+    assertFileExists $userConf
+    assertFileContent $userConf ${
+      pkgs.writeText "expected" ''
+        [Manager]
+        DefaultCPUAccounting=true
+        DefaultEnvironment=PATH='/bin:/sbin:/some where' TEST='abc'
+        LogLevel=debug
+        ManagerEnvironment=
```

Comment:
Is this problematic to include as an empty list?

Reply:
I wondered the same thing, but the [man page](https://www.freedesktop.org/software/systemd/man/latest/systemd.syntax.html) says (emphasis mine):

> and setting to an empty value "resets", which means that previous assignments are ignored. When this is allowed, **it is mentioned in the description of the setting**.

And there is no such mention on the [`systemd-user.conf`](https://www.freedesktop.org/software/systemd/man/latest/systemd-user.conf.html) man page.

However, having now run a test with `DefaultEnvironment`, it _does_ in fact reset when set to an empty value 🤦🏻.

Guess empty values have to filtered out then.

Reply:
Implemented filtering of empty/unset values.